### DISCUSSION TITLE
是怎么解析JSON后提交任务到flink的？说下大致逻辑

### DISCUSSION THREAD

**Discussion by wangztree** [DTStack/chunjun#1162] [Q&A]
是怎么解析JSON后提交任务到flink的？说下大致逻辑

**Comment by FlechazoW** [comment]
ChunJun是通过工具类，将json中的参数配置转换成内部的配置类，然后解析，通过SourceFactory、SinkFactory 构建flink 对应的PackagedProgram。
具体代码逻辑在Launcher，可以尝试debug下
